sql/core/src/main/scala/org/apache/spark/sql/execution/joins/CartesianProduct.scala

```diff
@@ -34,7 +34,15 @@ case class CartesianProduct(left: SparkPlan, right: SparkPlan) extends BinaryNod
     val leftResults = left.execute().map(_.copy())
     val rightResults = right.execute().map(_.copy())
 
-    leftResults.cartesian(rightResults).mapPartitions { iter =>
+    val cartesianRdd = if (leftResults.partitions.size > rightResults.partitions.size) {
+      rightResults.cartesian(leftResults).mapPartitions { iter =>
+        iter.map(tuple => (tuple._2, tuple._1))
+      }
+    } else {
+      leftResults.cartesian(rightResults)
+    }
+
+    cartesianRdd.mapPartitions { iter =>
       val joinedRow = new JoinedRow
```

Comment:
Quick question. Why not use the ```sizeInBytes```? I assume we want to move as little data as possible? Using ```sizeInBytes``` would be a bit more involved, since this would involve the planner, and (probably) adding a ```BuildSide``` parameter to ```CartesianProduct```...